packages/headless/src/features/facets/automatic-facet-set/automatic-facet-set-slice.ts

```diff
@@ -56,16 +57,45 @@ export const automaticFacetSetReducer = createReducer(
       .addCase(restoreSearchParameters, (state, action) => {
         const af = action.payload.af ?? {};
 
+        if (!state.set) {
+          state.set = {};
+        }
+
         for (const field in af) {
-          const response = buildTemporaryAutomaticFacetResponse(field);
-          const values = af[field].map((value) =>
-            buildTemporarySelectedFacetValue(value)
+          const response =
+            state.set[field]?.response ??
+            buildTemporaryAutomaticFacetResponse(field);
+
+          const selectedValues = af[field] || [];
+          const idleValues = response.values.filter(
+            (facetValue) => !selectedValues.includes(facetValue.value)
           );
-          response.values.push(...values);
+
+          response.values = [
+            ...selectedValues.map((value) => {
+              const facetValueObject = response.values.find(
+                (facetValue) => facetValue.value === value
+              );
+              const numberOfResults = facetValueObject?.numberOfResults;
+              return buildTemporarySelectedFacetValue(value, numberOfResults);
+            }),
+            ...idleValues.map(restoreFacetValueToIdleState),
+          ];
 
           state.set[field] = {response};
         }
       })
```

Comment:
Originally, I did not know that restoreSearchParameters would get called when going back and forward with the browser history. I only thought that restoreSearchParameters would get called when the page was reloading. I thought that the facet set would always be empty so I added bugs without knowing. This is the fix.